App review

williamolojede opened the pull request:
Review for the overall app.

williamolojede committed:
Merge pull request #12 from williamolojede/feature/user-login/15068756

test login route

williamolojede committed:
add recipe

williamolojede committed:
remove console log

williamolojede committed:
style guide double quote issue

williamolojede committed:
fix for missing file extension rule

williamolojede committed:
Merge pull request #13 from williamolojede/feature-add-recipe-150575629

#150575629 add recipe

williamolojede committed:
add delete recipe route with tests

williamolojede committed:
remove console log

williamolojede committed:
fix file name

williamolojede committed:
fix travis tes command

williamolojede committed:
remove node iojs version in travis

williamolojede committed:
Merge pull request #14 from williamolojede/feature/delete-recipe-route/150575680

#150575680 add delete recipe route with tests

williamolojede committed:
add recipe

williamolojede committed:
Merge pull request #15 from williamolojede/feature/get-all-recipes/150761566

#150761566 add recipe

williamolojede committed:
add route to allow user modify recipe

williamolojede committed:
Merge pull request #16 from williamolojede/feature/modify-a-recipe/150575632

##150575632 route to allow logged in user modify recipe

williamolojede committed:
add associations for votes and recipe

williamolojede committed:
vote on a recipe

williamolojede committed:
Merge pull request #17 from williamolojede/feature/vote-on-recipe/150766083

#150766083 vote on recipe

williamolojede committed:
add associations for user, favorite and recipe

williamolojede committed:
add route for favorite a recipe with test

williamolojede committed:
removed unsed variable --eslint rule

williamolojede committed:
Merge pull request #18 from williamolojede/feature/favorite-a-recipe/150767360

#150767360 favorite a recipe

williamolojede committed:
add get user's favorite recipe route with tests

williamolojede committed:
Merge pull request #19 from williamolojede/feature/get-user-favorite/150575639

#150575639 get user's favorite recipe

williamolojede committed:
get recipes with the most upvotes

williamolojede committed:
Merge pull request #20 from williamolojede/feature/get-top-upvotes/150575642

#150575642 get recipes with the most up-votes

williamolojede committed:
refactor old code and add test for post reviw

williamolojede committed:
add route to allow user post review

williamolojede committed:
add procfile

williamolojede committed:
add github pr template

williamolojede committed:
Merge pull request #21 from williamolojede/feature/post-review/150761581

# 150761581 post review

williamolojede committed:
configure travis coverage

williamolojede committed:
fix coverage command

williamolojede committed:
fix coverage command

williamolojede committed:
fix coverage command

williamolojede committed:
add codeclimate config

williamolojede committed:
fix coverage commands

williamolojede committed:
fix coverage commands

williamolojede committed:
fix coverage commands

williamolojede committed:
change travis after script

williamolojede committed:
fix coverage

williamolojede committed:
fix coverage

williamolojede committed:
fix travis tes command

williamolojede committed:
fix travis

williamolojede committed:
travis

williamolojede committed:
travis

williamolojede committed:
travis

williamolojede committed:
travis finally

williamolojede committed:
codeclimate

williamolojede committed:
add codeclimate badges

williamolojede committed:
Merge pull request #22 from williamolojede/chore/host-on-heroku/150833896

Chore/host on heroku/150833896

williamolojede committed:
ignore test path for codeclimate

williamolojede committed:
Merge pull request #23 from williamolojede/chore/host-on-heroku/150833896

ignore test path for codeclimate

williamolojede committed:
inital heroku setup

williamolojede committed:
Merge pull request #24 from williamolojede/chore/host-on-heroku/150833896

inital heroku setup

williamolojede committed:
fix catch block

williamolojede committed:
add postbuild script for heroku to run migrate

williamolojede committed:
Merge pull request #25 from williamolojede/chore/host-on-heroku/150833896

# 150650156 fix catch block

williamolojede committed:
fix counting method

williamolojede committed:
remove comments

williamolojede committed:
refactor review create function

williamolojede committed:
Merge pull request #26 from williamolojede/feature/get-all-recipes/150761566

fix counting method

williamolojede committed:
remove unsed npm script

williamolojede committed:
add test for the descriped scenerio

williamolojede committed:
fix bug #150909727

williamolojede committed:
Merge pull request #27 from williamolojede/bug/fix-review-response/150909727

#150909727 fix review response

williamolojede committed:
remove console log from test file

williamolojede committed:
test for error file

williamolojede committed:
fix travis test command

williamolojede committed:
add test for expired token validation

williamolojede committed:
increase mocha test duration limit

williamolojede committed:
fix style guide error

williamolojede committed:
Merge pull request #28 from williamolojede/bug/fix-review-response/150909727

test for system error handler

williamolojede committed:
add endpoints to readme

williamolojede committed:
Merge pull request #29 from williamolojede/chore/write-readme/150925173

add endpoints to readme

williamolojede committed:
add route to get a single recipe

williamolojede committed:
Merge pull request #30 from williamolojede/feature/get-single-recipe-details/150761581

#150761581 add route to get a single recipe

williamolojede committed:
add model for views table

williamolojede committed:
add middleware to count a recipe's view

williamolojede committed:
ignore model directory in codeclimate

williamolojede committed:
delete isowner middleware

williamolojede committed:
Merge pull request #31 from williamolojede/feature/implement-view-count/151120458

#151120458 implement view count

williamolojede committed:
validate signup

williamolojede committed:
validate recipe creation

williamolojede committed:
add extra test for updating recipe

williamolojede committed:
remove wrong file

williamolojede committed:
fix validation for writing review

williamolojede committed:
add extra test to cover line 40 of reviewRecipe.js

williamolojede committed:
refactor middleware import

williamolojede committed:
refactor middleware import

williamolojede committed:
Merge pull request #32 from williamolojede/chore/route-input-validation/150943197

#150943197 route input validation

williamolojede committed:
refactor sigup request body

williamolojede committed:
refactor login request body

williamolojede committed:
refactor recipe request body

williamolojede committed:
refactor login validation

williamolojede committed:
refactor signup validation

williamolojede committed:
Merge pull request #33 from williamolojede/chore/refactor-request-body/151151113

#151151113 refactor request body

williamolojede committed:
initial setup

williamolojede committed:
ignore dist directory

williamolojede committed:
add sass file

williamolojede committed:
add image assets

williamolojede committed:
setup express serve to serve static files

williamolojede committed:
setup express serve to serve static files

williamolojede committed:
configure production build and hot reloading

williamolojede committed:
Merge pull request #34 from williamolojede/chore/react-development-setup/151159425

#151159425 react development setup

williamolojede committed:
fix heroku issue

williamolojede committed:
Merge pull request #35 from williamolojede/heroku-fix

fix heroku issue

williamolojede committed:
fix dependecies

williamolojede committed:
Merge pull request #36 from williamolojede/heroku-fix

fix dependecies

williamolojede committed:
move babel load to dependencies

williamolojede committed:
heroku fix

williamolojede committed:
css loader

williamolojede committed:
fix

williamolojede committed:
Merge pull request #37 from williamolojede/heroku-fix

Heroku fix

williamolojede committed:
setup doc hosting

williamolojede committed:
fix docs production build script

williamolojede committed:
change docs build location

williamolojede committed:
add api documentation

williamolojede committed:
add eslint ignore

williamolojede committed:
update readme with links to api doc and web client

williamolojede committed:
Merge pull request #38 from williamolojede/chore/write-api-doc/151101973

Chore/write api doc/151101973

williamolojede committed:
add login view

williamolojede committed:
add reducers and actions for loginaction

williamolojede committed:
WIP: implement submit login form

williamolojede committed:
implement redirect for protected routes and login view

williamolojede committed:
change token payload

williamolojede committed:
remove console.log statement

williamolojede committed:
Merge pull request #39 from williamolojede/chore/change-user-token/151505740

#151505740 change token payload

williamolojede committed:
update branch with current remote dev branch

williamolojede committed:
ensure that user data and token is in store

williamolojede committed:
Merge pull request #40 from williamolojede/feature/login-view/150468875

#150468875 login view

williamolojede committed:
refactored is fetching

williamolojede committed:
remove in-file action type declaration

williamolojede committed:
connect to backend

williamolojede committed:
Merge pull request #41 from williamolojede/signup-view

Signup view

williamolojede committed:
add homepage layout no functionality

williamolojede committed:
add fetching recipes from api endpoint

williamolojede committed:
abstract all proptypes definition

williamolojede committed:
add materialize js

williamolojede committed:
add view for when api returns no recipes

williamolojede committed:
add logic to check if user token as expired

williamolojede committed:
Merge pull request #42 from williamolojede/feature/homepage-view/150469037

#150469037 homepage view

williamolojede committed:
WIP

williamolojede committed:
change db username

williamolojede committed:
alot

williamolojede committed:
implement single recipe page actions

williamolojede committed:
Merge pull request #43 from williamolojede/feature/single-recipe-view/150469146

Feature/single recipe view/150469146

williamolojede committed:
implement add get user detail

williamolojede committed:
add validation if uid provided does not match any user in db

williamolojede committed:
add test for get user

williamolojede committed:
add extra test expectation

williamolojede committed:
Merge pull request #44 from williamolojede/feature/get-user-details/151150801

Feature/get user details/151150801

williamolojede committed:
feature(user profile) add user profile page

- the user can view his/her personal recipes or a list of favorite ones
also a user can add new recipes, update/delete existing ones and remove
a recipe from the favorite recipe list

williamolojede committed:
Merge pull request #45 from williamolojede/feature/user-profile-view/150469557

#150469557 add user profile page

williamolojede committed:
bug(authentication) fix error message notification

previously, the error message remains after been shown and also when
moving from login to signup page and vice versa this commit fixes that

williamolojede committed:
Merge pull request #46 from williamolojede/bug/error-notification/152462701

#152462701  fix error message notification

williamolojede committed:
bug(homepage) add call to action message to homepage

- add search icon to search input on homepage
- add call to action message

williamolojede committed:
Merge pull request #47 from williamolojede/bug/hero-section-home-page/152500889

#152500889 hero section on homepage

williamolojede committed:
feature(authentication) add sign out button

Add a link on the navbar to allow users sigout of the application

williamolojede committed:
Merge pull request #48 from williamolojede/feature/152533025/logout

#152533025 sign out

williamolojede committed:
feature(single recipe page) separete rows

separate ingredients and instructions list to separate rows

williamolojede committed:
Merge pull request #49 from williamolojede/feature/152539819/instructions-ingredients-list

#152539819 ingredients and instructions list on separete rows

williamolojede committed:
bug(single recipe):  change loading screen to notification

- change loading screen when user votes or favorite a recipe to a
notification
[Delivers #152570242]

williamolojede committed:
Merge pull request #50 from williamolojede/bug/152570242/vote-favoriting

bug(single recipe):  change loading screen to notification

williamolojede committed:
feature(homepage) - pagination

williamolojede committed:
Merge pull request #51 from williamolojede/feature/152721187/homepage-pagination

pagination on homepage

williamolojede committed:
bug(userImg) - fix code breaking when user fullname is just one word

williamolojede committed:
Merge pull request #52 from williamolojede/bug/152860238/user-default-image

fix code breaking when user fullname is just one word

williamolojede committed:
bug(api/client) fix error when adding review and recipe

williamolojede committed:
Merge pull request #53 from williamolojede/bug/152883835/error-adding-review-recipe

fix error when adding review and recipe

williamolojede committed:
feature(profile) - add pagination for personal and favorite recipe

williamolojede committed:
Merge pull request #54 from williamolojede/feature/152957152/pagination-user-profile-page

pagination for profile page

williamolojede committed:
feature(navbar) - make hamburger button work on mobile

williamolojede committed:
feature(navbar) - make hamburger button work on mobile

williamolojede committed:
Merge pull request #55 from williamolojede/feature/152995512/mobile-navigation

mobile navigation

williamolojede committed:
Merge pull request #56 from williamolojede/feature/152995512/mobile-navigation

mobile navigation

williamolojede committed:
feature(search) - implement search on the homepage

williamolojede committed:
Merge pull request #57 from williamolojede/feature/150575582/homepage-search

search

BolajiOlajide commented on .coveralls.yml:
```diff
@@ -1 +1,2 @@
-repo_token: ei2zLuDO3en8g6fZb6QthEWAoPveYjpwr
+service_name: travis-ci
+repo_token: 0wlHHzTif1fIqHHNRie6cxo7ecb7woebY
```
You might want to hide this token, check out how to make use of encrypt your token with Travis [here](https://docs.travis-ci.com/user/encryption-keys/)

phabbs replied:
I think the token should be stored in env variable not pushed to public repo

BolajiOlajide commented on .env.sample:
```diff
@@ -0,0 +1,4 @@
+FIREBASE_API_KEY=
+FIREBASE_AUTH_DOMAIN=
+FIREBASE_DB_URL=
```
Are you sure this is all the environment variable needed to setup your application?
I think somethings are missing, not sure though.

BolajiOlajide commented on .travis.yml:
```diff
@@ -1,18 +1,29 @@
+env:
+  global:
+    - CC_TEST_REPORTER_ID: "e895d45cfaffba8fb9a9fb36c8b3175563b0292e639308921059da19bf4fc5d2"
```
You might want to protect all this variables you're revealing to the public. Hackers are real.

BolajiOlajide commented on README.md:
```diff
@@ -1,5 +1,24 @@
+ 1. Express
+ 1. Sequelize
+ 1. React
+ 1. Redux
```
Other technologies like Mocha or Jest, enzyme and the rest are also worth mentioning.

BolajiOlajide commented on client/components/App.jsx:
```diff
@@ -0,0 +1,39 @@
+    return (
+      <BrowserRouter>
+        <Switch>
+          <Route exact path="/" component={requireAuthentication(Home)} />
```
I feel you're repeating yourself here with the `requireAuthentication` method.
There ought to be a simpler way to do this.

phabbs commented on client/actions/fetchRecipe.js:
```diff
@@ -0,0 +1,162 @@
+import { batchActions } from 'redux-batched-actions';
```
I think this file should be name `recipe.js` instead of `fetchRecipe.js` because it contains other actions that does not have anything to do with fetching recipe

phabbs commented on client/actions/fetchRecipe.js:
```diff
@@ -0,0 +1,162 @@
+  type: RECIEVE_NEW_RECIPE,
+  recipe
+});
+
```
I think recieveSingleRecipe and recieveSingleRecipe can be merge to =>

const recieveSingleRecipe = (type, recipe) => ({
   type,
   recipe
 });

phabbs commented on client/actions/fetchRecipe.js:
```diff
@@ -0,0 +1,162 @@
+      );
+    })
+    .catch((err) => {
+      console.log(err.message || err.response.data.message);
```
I think the proper way to display error is through snackbar or other forms of notifications as opposed to logging it.

azeezolaniran2016 replied:
IMO, logging should be for debugging so this is fine especially if it was possible to toggle debug mode. Still, you should display some notification to the user if this error is something they should be aware of.

phabbs commented on client/actions/fetchRecipe.js:
```diff
@@ -0,0 +1,162 @@
+      );
+    })
+    .catch((err) => {
+      console.log(err.message || err.response.data.message);
```
Improve handling of error

phabbs commented on client/actions/fetchRecipe.js:
```diff
@@ -0,0 +1,162 @@
+      );
+    })
+    .catch((err) => {
+      console.log(err.response.data.message || err.message);
```
Improve handling of error

phabbs commented on client/actions/fetchRecipe.js:
```diff
@@ -0,0 +1,162 @@
+    dispatch(showNotification(err.response.data.message));
+    setTimeout(() => {
+      dispatch(hideNotification());
+    }, 3000);
```
### Nitpick
I think its better to handle notifications and its timing in the components

phabbs commented on client/actions/fetchRecipe.js:
```diff
@@ -0,0 +1,162 @@
+      );
+    })
+    .catch((err) => {
+      console.log(err.message || err.response.data.message);
```
Improve error handling

phabbs commented on client/actions/loginUser.js:
```diff
@@ -0,0 +1,44 @@
+import { LOGIN_SUCCESS, LOGIN_FAILURE, LOGOUT_SUCCESS } from './types';
+
+const receiveLogout = () => ({
+  type: LOGOUT_SUCCESS,
```
### Critical
Since nothing is received, I think the name `receiveLogout` is misleading

phabbs commented on client/actions/loginUser.js:
```diff
@@ -0,0 +1,44 @@
+import { batchActions } from 'redux-batched-actions';
+
+import { setFetching, unsetFetching } from './fetching';
+import { receiveAuth, handleAuthSuccess, handleAuthFailure } from './auth';
```
### Critical
Is there a reason the actions in the `../actions/auth` and those in `../actions/loginUser` can not be in the same file?

williamolojede replied:
that's because `auth.js` contains methods that are common to both `loginUser.js` and `signupUser.js`.

phabbs commented on client/actions/signupUser.js:
```diff
@@ -0,0 +1,27 @@
+      email: creds.email,
+      password: creds.password,
+      fullname: creds.fullname
+    }
```
### Critical
You can rewrite this as;
```
const body = {
    user: cred
  }
```

femisystems commented on .travis.yml:
```diff
@@ -1,18 +1,29 @@
 
+addons:
+  code_climate:
+    repo_token: "e895d45cfaffba8fb9a9fb36c8b3175563b0292e639308921059da19bf4fc5d2"
```
`repo_token` can be included in travis environment.

azeezolaniran2016 commented on client/actions/fetchRecipe.js:
```diff
@@ -0,0 +1,162 @@
+    }, 3000);
+  })
+  .catch((err) => {
+    dispatch(showNotification(err.response.data.message));
```
Are you sure you can depend on this error message to always be something sensible to the user?

williamolojede committed:
chore(env) - add environment vairables and refactor test

williamolojede committed:
Merge pull request #59 from williamolojede/chore/153114990/env-variables

#153114990 environment vairables

williamolojede committed:
implement feedback and fix hound errors

williamolojede committed:
implement feedback and fix hound errors

williamolojede committed:
Merge pull request #60 from williamolojede/feedback-client-side

implement feedback and fix hound errors

williamolojede committed:
Merge branch 'dev' into feedback-client-side

williamolojede committed:
Merge branch 'dev' into feedback-client-side

williamolojede committed:
Merge pull request #61 from williamolojede/feedback-client-side

implement feedback and fix hound errors

williamolojede committed:
Merge branch 'feedback-client-side' of github.com:williamolojede/more-recipes into feedback-client-side

williamolojede committed:
Merge branch 'feedback-client-side' of github.com:williamolojede/more-recipes into feedback-client-side

williamolojede committed:
Merge pull request #62 from williamolojede/feedback-client-side

Feedback client side

williamolojede committed:
Merge pull request #63 from williamolojede/feedback-client-side

feedback-client-side

williamolojede committed:
replace the name for currentValue of looping through an array

williamolojede committed:
make api response body(status, message) consistent

williamolojede committed:
set default(and refactor) pagination

williamolojede committed:
refactor validation for id parameters(users, recipes)

williamolojede committed:
refactor and rename module

williamolojede committed:
refactor login controller

williamolojede committed:
refactor token creation and current user on client side

williamolojede committed:
Refactor, Pagination, New Endpoint

 - Add Pagination to all endpoints that returns a list of items
 - Refactor default props on client side
 - Refactor Reducers
 - Add endpoints to get list of user's own recipes

williamolojede committed:
update readme

williamolojede committed:
update readme

williamolojede committed:
update docs

williamolojede committed:
update codeclimate config

williamolojede committed:
update http status code

williamolojede committed:
update tests

williamolojede committed:
update docs

williamolojede committed:
Merge pull request #64 from williamolojede/defence1-feedback

Defence1 feedback

williamolojede committed:
restructure component folder and fix issues

williamolojede committed:
Merge pull request #65 from williamolojede/chore/154305083/component-folder-restructure

restructure component folder and fix issues

williamolojede committed:
Implement pagination for reviews and add test

williamolojede committed:
Merge pull request #66 from williamolojede/feature/154305050/reviews-ui-pagination

Implement pagination for reviews

williamolojede committed:
Create a dedicated page for adding and managing user recipes

- Also add client side test

williamolojede committed:
Merge pull request #67 from williamolojede/feature/154358921/dedicated-recipe-management-page

Create a dedicated page for adding and managing user recipes

williamolojede committed:
implement not found page

williamolojede committed:
Merge pull request #68 from williamolojede/feature/154521114/not-found

implement not found page

williamolojede committed:
implement unique recipe name

williamolojede committed:
Merge pull request #69 from williamolojede/feature/154522067/unique-recipe-name-per-user

implement unique recipe name

williamolojede committed:
Add tests and Update docs

- End-to-End tests
- Models Tests
- Refactor Api Test(split into separate files)
- Update docs

williamolojede committed:
Merge pull request #70 from williamolojede/chore/154522067/e2e-testing

Add tests and Update docs

williamolojede committed:
add support for cors
